lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -0,0 +1,188 @@
+          'delivering critical JS/CSS inline and deferring all non-critical ' +
+          'JS/styles. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/blocking-resources).',
+      requiredArtifacts: ['CSSUsage', 'URL', 'TagsBlockingFirstPaint', 'traces'],
+    };
+  }
+
+  /**
+   * @param {Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   */
+  static async computeResults(artifacts, context) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const simulatorData = {devtoolsLog, settings: context.settings};
+    const traceOfTab = await artifacts.requestTraceOfTab(trace);
+    const simulator = await artifacts.requestLoadSimulator(simulatorData);
+    const wastedBytesMap = await RenderBlockingResources.computeWastedCSSBytes(artifacts, context);
+
+    const metricSettings = {throttlingMethod: 'simulate'};
+    const metricComputationData = {trace, devtoolsLog, simulator, settings: metricSettings};
+    const fcpSimulation = await artifacts.requestFirstContentfulPaint(metricComputationData);
+    const fcpTsInMs = traceOfTab.timestamps.firstContentfulPaint / 1000;
+
+    const nodeTimingMap = fcpSimulation.optimisticEstimate.nodeTiming;
+    const nodesByUrl = keyByUrl(Array.from(nodeTimingMap.keys()));
+
+    const results = [];
+    const deferredNodeIds = new Set();
+    for (const resource of artifacts.TagsBlockingFirstPaint) {
+      // Ignore any resources that finished after observed FCP (they're clearly not render-blocking)
+      if (resource.endTime * 1000 > fcpTsInMs) continue;
+
+      const node = nodesByUrl[resource.tag.url];
+      const nodeTiming = nodeTimingMap.get(node);
+      // TODO(phulce): beacon these occurences to Sentry to improve FCP graph
+      if (!node) continue;
+
+      // Mark this node and all it's dependents as deferrable
+      // TODO(phulce): make this slightly more surgical
+      // i.e. the referenced font asset won't become inlined just because you inline the CSS
+      node.traverse(node => deferredNodeIds.add(node.id));
+
+      const wastedMs = nodeTiming.endTime - nodeTiming.startTime;
+      if (wastedMs < MINIMUM_WASTED_MS) continue;
+
+      results.push({
+        url: resource.tag.url,
+        totalBytes: resource.transferSize,
+        wastedMs,
+      });
+    }
+
+    if (!results.length) {
+      return {results, wastedMs: 0};
+    }
+
+    const wastedMs = RenderBlockingResources.estimateSavingsFromInlining(
+      simulator,
+      fcpSimulation.optimisticGraph,
+      deferredNodeIds,
+      wastedBytesMap
+    );
+
+    return {results, wastedMs};
+  }
+
+  /**
+   * @param {Simulator} simulator
+   * @param {Node} fcpGraph
+   * @param {Set<string>} deferredIds
+   * @param {Map<string, number>} wastedBytesMap
+   * @return {number}
+   */
+  static estimateSavingsFromInlining(simulator, fcpGraph, deferredIds, wastedBytesMap) {
```

Comment:
also since the inlining only applies to CSS but this affects both JS and CSS it should be named something that's more balanced.  `estimateSavingsFromDeferring` sg. wdyt?

Reply:
since it's doing both, I just renamed to `estimateSavingsWithGraphs` and added better description 👍